go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -443,28 +444,99 @@ func handleNetdevResources(resourceName string) (string, error) {
 	return netdevice, nil
 }
 
-func createNodeManagementPorts(name string, nodeAnnotator kube.Annotator, waiter *startupWaiter,
-	subnets []*net.IPNet) ([]managementPortEntry, *managementPortConfig, error) {
-	// If netdevice name is not provided in the full mode then management port backed by OVS internal port.
-	// If it is provided then it is backed by VF or SF and need to determine its representor name to plug
-	// into OVS integrational bridge
+func exportManagementPortAnnotation(netdevName string, nodeAnnotator kube.Annotator) error {
+	klog.Infof("Exporting management port annotation")
+
+	deviceID, err := util.GetDeviceIDFromNetdevice(netdevName)
+	if err != nil {
+		return err
+	}
+	vfindex, err := util.GetSriovnetOps().GetVfIndexByPciAddress(deviceID)
+	if err != nil {
+		return err
+	}
+	pfindex, err := util.GetSriovnetOps().GetPfIndexByVfPciAddress(deviceID)
+	if err != nil {
+		return err
+	}
+
+	return util.SetNodeManagementPortAnnotation(nodeAnnotator, pfindex, vfindex)
+}
+
+func importManagementPortAnnotation(node *kapi.Node) (string, error) {
+	klog.Infof("Import management port annotation")
+	pfId, vfId, err := util.ParseNodeManagementPortAnnotation(node)
+
+	if err != nil {
+		return "", err
+	}
+
+	return sriovnet.GetVfRepresentorDPU(fmt.Sprintf("%d", pfId), fmt.Sprintf("%d", vfId))
+}
+
+func getManagementPortNetDev(netdevName string) (string, error) {
+	link, err := util.GetNetLinkOps().LinkByName(netdevName)
+	if err != nil {
+		if !util.GetNetLinkOps().IsLinkNotFoundError(err) {
+			return "", fmt.Errorf("failed to lookup %s link: %v", netdevName, err)
+		}
+		// this may not the first time invoked on the node after reboot
+		// netdev may have already been renamed to ovn-k8s-mp0.
+		link, err = util.GetNetLinkOps().LinkByName(types.K8sMgmtIntfName)
+		if err != nil {
+			return "", fmt.Errorf("failed to get link device for %s. %v", netdevName, err)
+		}
+	}
+	return link.Attrs().Name, err
+}
+
+
+// If netdevice name is not provided in the full mode then management port backed 
+// by OVS internal port. If it is provided then it is backed by VF or SF and need
+// to determine its representor name to plug into OVS integrational bridge
+// In case ovn-k is running in DPU mode, it will read the annotation from the host
+// side which should have been exported by ovn-k running in DPU host mode.
+func getMgmtPortAndRepName(node *kapi.Node) (string, string, error) {
 	if config.OvnKubeNode.Mode == types.NodeModeFull && config.OvnKubeNode.MgmtPortNetdev != "" {
-		deviceID, err := util.GetDeviceIDFromNetdevice(config.OvnKubeNode.MgmtPortNetdev)
+		var err error
+		netdevName := config.OvnKubeNode.MgmtPortNetdev
+		netdevName, err = getManagementPortNetDev(netdevName)
```

Comment:
I see because `config.OvnKubeNode.MgmtPortNetdev` maybe an alternative name and we need `getManagementPortNetDev()` to make sure we get the correct interface. Is this correct @DmytroLinkin ?

Reply:
Yes, this is one fo the reasons it actually happens to work in OCP. In the first run, the port gets renamed. The next run, it still picks up the port and renames it again.

Reply:
> I see because `config.OvnKubeNode.MgmtPortNetdev` maybe an alternative name and we need `getManagementPortNetDev()` to make sure we get the correct interface. Is this correct @DmytroLinkin ? Nice catch.

No. It is opposite. When no altnames are used. E.g., `ip link show dev <iface>` doesn't show altnames in the output.